cli/internal/cmd/workspace/workspace.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+/*
+Package workspace provides functions to generate paths relative to the workspace.
+
+Users may override the default workspace using the --workspace flag.
+The default workspace is the current working directory.
+
+The functions defined here return paths relative to the set workspace,
+and should be used when printing the path to the user.
+
+The MUST not be used when accessing files, as the workspace is changed
+using os.Chdir() before the command is executed.
```

Comment:
`return paths relative to the set workspace` this is not true, as far I understand it. They return the path to the actual pwd (before the -C execution). 

To avoid misuse, should we omit this functionality entirely and print the paths in relation to the set workdir? How do other tools behave here? I always understood `-C` as literally `cd` into the folder but if the program returns, I want to be back in the original path. I'd assume that e.g. `make` does not "normalize" the paths as we do here.

Reply:
>I always understood -C as literally cd into the folder

Correct. One exit, you will be back where you started.
However, we have multiple cases where we print filepaths to the user for 
For example, after a successful init we print the path to the kubeconfig file:

```
Constellation ID file written to <workspace>/constellation-admin.conf
You can now connect to your cluster by executing:
    export KUBECONFIG="$PWD/<workspace>/constellation-admin.conf
```
If we omit this package, those paths will not work correctly.

(Looking at this again, this path may also not be printed correctly when one provides an absolute path. Should be easy to fix though.)

Reply:
> However, we have multiple cases where we print filepaths to the user for

What are the other cases? Can we simply always print the absolute path?

Reply:
> What are the other cases?

Usually any time we create or update files, e.g. during config generate, and upgrade apply/check. It is also required for printing error messages to the user.
We also print a lot of filenames for debug statements.

> Can we simply always print the absolute path?

In that case we would always have to call `filepath.Abs`, so there is no real advantage gained, other than not parsing the workspace flag, with the downside of paths looking less pretty.

I have also implemented the suggestion from https://github.com/edgelesssys/constellation/pull/2181#discussion_r1287386166, which should make usage of this package a lot easier.